cli/server/server.go

```diff
@@ -528,20 +528,14 @@ Main:
 			switch sig {
 			case syscall.SIGHUP:
 				log.Info("SIGHUP received, restarting rpc-server")
-				serverErr := rpcServer.Shutdown()
-				if serverErr != nil {
-					errChan <- fmt.Errorf("error while restarting rpc-server: %w", serverErr)
-					break
-				}
-				rpcServer = server.New(chain, cfg.ApplicationConfiguration.RPC, serv, oracleSrv, log)
-				rpcServer.Start(errChan)
+				rpcServer.Shutdown()
```

Comment:
It seems there is a potential deadlock here, because we have an unbuffered channel which is read in the same `select`. Am I missing something?

Reply:
That very well might be the case, although nothing changed wrt it in this PR.

Reply:
Fixed.